tmt/utils/__init__.py

```diff
@@ -4328,11 +4335,42 @@ def send(  # type: ignore[override]
 
 
 class RetryStrategy(urllib3.util.retry.Retry):
+    def __init__(self, *args: Any, logger: Optional[tmt.log.Logger] = None, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.logger = logger
+
+    def new(self, **kw: Any) -> 'Self':
+        new_retry = super().new(**kw)
+        new_retry.logger = self.logger
+        return new_retry
+
+    def _log_rate_limit_info(self, headers: urllib3._collections.HTTPHeaderDict) -> None:
+        """Log current GitHub rate limit information"""
+        if not self.logger:
+            return
+
+        self.logger.debug("Limit", headers.get('X-RateLimit-Limit', 'unknown'))
+        self.logger.debug("Remaining", headers.get('X-RateLimit-Remaining', 'unknown'))
+        self.logger.debug("Reset", headers.get('X-RateLimit-Reset', 'unknown'))
+        self.logger.debug("Resource", headers.get('X-RateLimit-Resource', 'unknown'))
+
+    def _log_response_info(self, response: HTTPResponse) -> None:
+        """Log detailed response information for debugging"""
+        if not self.logger:
+            return
+
+        self.logger.debug("Response status", response.status)
+        self.logger.debug("Response headers", dict(response.headers))
+        self.logger.debug("Response text", response.data.decode('utf-8'))
+
     def increment(
             self,
             *args: Any,
             **kwargs: Any
             ) -> urllib3.util.retry.Retry:
+        if self.logger:
+            self.logger.debug(f'increment: {args=} {kwargs=}')
```

Comment:
This call should be dropped, it was in my patch just to trace those calls, but it's not needed.

Reply:
done